integrations/airflow/marquez_airflow/dag.py

```diff
@@ -125,60 +130,61 @@ def handle_callback(self, *args, **kwargs):
 
         return super().handle_callback(*args)
 
-    def _report_task_instances(self, dagrun, run_args, session):
+    def _report_task_instances(self, dagrun, session):
         task_instances = dagrun.get_task_instances()
-        for ti in task_instances:
+        for task_instance in task_instances:
             try:
-                self._report_task_instance(ti, dagrun, run_args, session)
+                self._report_task_instance(task_instance, dagrun, session)
             except Exception as e:
                 self.log.error(
                     f'Failed to record task instance: {e} '
                     f'dag_id={self.dag_id}',
                     exc_info=True)
 
-    def _report_task_instance(self, ti, dagrun, run_args, session):
-        task = self.get_task(ti.task_id)
-        run_ids = JobIdMapping.pop(
-            self._marquez_job_name_from_ti(ti), dagrun.run_id, session)
-        steps = self._extract_metadata(dagrun, task, ti)
+    def _report_task_instance(self, task_instance, dagrun, session):
+        task = self.get_task(task_instance.task_id)
 
-        # Note: run_ids could be missing if it was removed from airflow
+        # Note: task_run_id could be missing if it was removed from airflow
         # or the job could not be registered.
```

Comment:
The danger here is in some sub-workflow that registered itself as a child of this task losing its connection if the original DAG task never completes, right? Can we consider deriving deterministic task run ids so that if we somehow lose the job id mapping, we can still generate the same id? E.g., I think the DAG run id is already a UUID, right? Is there harm in appending the task name to the DAG run id and generating a unique run id that way?